.goreleaser.yml

```diff
@@ -0,0 +1,84 @@
+# This is an example .goreleaser.yml file with some sane defaults.
+# Make sure to check the documentation at http://goreleaser.com
+before:
+  hooks:
+    - go mod tidy
+builds:
+  - env:
+      - CGO_ENABLED=0
+    goos:
+      - linux
+      - windows
+      - darwin
+    goarch:
+      - amd64
+    main: ./cmd/imgpkg
+    binary: imgpkg-{{ .Os }}-{{ .Arch }}
+
+    # not sure if the ldflags buildid is needed anymore. check https://github.com/golang/go/issues/16860
+    flags:
+      - -trimpath
+
+    ldflags:
+      - -buildid=
+      - -X github.com/k14s/imgpkg/pkg/imgpkg/cmd.Version={{ .Version }}
```

Comment:
what are thoughts behind injecting version here vs in code?

Reply:
It results in 1 less manual step done during a release.

Instead of making a code change and then pushing a tag. It changes to: 'push a tag'

If someone wanted to checkout the code and build our project, they can rebuild the *same* binary uploaded via `goreleaser` or via the `hack/build-binary.sh`.